src/app/clusters/door-lock-server/door-lock-server-core.cpp

```diff
@@ -76,7 +76,7 @@ static void setLanguage(void)
     }
 }
 
-void emberAfPluginDoorLockServerInitCallback(void)
+extern "C" void emberAfPluginDoorLockServerInitCallback(void)
```

Comment:
Why don't we have a declaration for `emberAfPluginDoorLockServerInitCallback` in an `#include`d header?

Reply:
The declaration of `emberAfPluginDoorLockServerInitCallback` comes from one of the generated files from Simplicity Studio, in this case https://github.com/project-chip/connectedhomeip/blob/fbbf127d835f08d5304de446da38c16d79014ee7/examples/wifi-echo/server/esp32/main/gen/endpoint_config.h#L406

I can not really alter the generated code. But all of this would be easier once the `gen/` files comes from ZAP and all of `src/app` has been converted to cpp.

Because of this, I'm re-requesting review.